src/Native/System.Net.Http.Native/pal_easy.h

```diff
@@ -142,3 +149,62 @@ Unpauses the CURL request.
 Returns CURLE_OK (0) if everything was ok, non-zero means an error occurred.
 */
 extern "C" int32_t EasyUnpause(CURL* handle);
+
+// A "base type" for the below function pointers, so they can be held generically
+// by the CallbackHandle type.
+typedef void(*FunctionPointer)();
+
+// the function pointer definition for the callback used in RegisterSeekCallback
+typedef int32_t(*SeekCallback)(void* userPointer, int64_t offset, int32_t origin);
+
+// the function pointer definition for the callback used in RegisterReadWriteCallback
+typedef uint64_t(*ReadWriteCallback)(uint8_t* buffer, uint64_t bufferSize, uint64_t nitems, void* userPointer);
+
+// the function pointer definition for the callback used in RegisterSslCtxCallback
+typedef int32_t(*SslCtxCallback)(CURL* curl, void* sslCtx);
+
+/*
+The object that is returned from RegisterXXXCallback functions.
+This holds the data necessary to know what managed callback to invoke and with what args.
+*/
+struct CallbackHandle
+{
+    FunctionPointer callback;
+    void* userPointer;
+};
+
```

Comment:
Yes, the layout is opaque to managed code. I will move the definition of this struct to the .cpp file.